src/OpenTelemetry.AutoInstrumentation.Native/stack_capture_strategy.h

```diff
@@ -0,0 +1,84 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+#include <corhlpr.h>
+#include <corprof.h>
+#include <unordered_set>
+namespace continuous_profiler {
+
+/// <summary>
+/// Platform-agnostic interface for capturing thread stacks.
+/// Different implementations handle platform-specific suspension mechanisms:
+/// - .NET Core/5+: CLR suspension (SuspendRuntime/ResumeRuntime)
+/// - .NET Framework: Per-thread suspension + seeded DoStackSnapshot
+/// </summary>
+
+    using StackSnapshotCallbackRaw = HRESULT(*)(
+        FunctionID funcId,
+        UINT_PTR ip,
+        COR_PRF_FRAME_INFO frameInfo,
+        ULONG32 contextSize,
+        BYTE context[],
+        void* clientData);
```

Comment:
It is hard to understand, as we have to many wrappers that casted to `void*`. As that function is internal use only, it may help to declare type of `clientData*` here explicitly - even if it means to use forward declaration.
Are we not using `std::function` wrapper here due to perfomance cost? Otherwise it may have requred internal clientData in closure, instea of passing it outside.